### ISSUE TITLE
Save should accept string or array in order to do PATCH instead of PUT

### ISSUE THREAD

**Issue by donaldpipowitch** [mgonto/restangular#722]
In #584 you decided against PATCH support within `.save()`, because you said you would need to keep a copy of the complete object in `$rootScope`. But sometimes you know beforehand that some elements are just a _partial element_. Say you have a REST API like `/api/users/123?fields=name,age` than you get some object like `{ "name": "Pipo", "age": 25 }` and you know that the complete element would have even more properties. It would be great to use PATCH instead of PUT within `.save()` then.

**Comment by grabbou** [comment]
I like idea of passing array of properties or a string of single property to be passed which will execute PATCH request instead of PUT. Will throw an error if ID is empty.

``js
user.save('name');
/ PATCH /users {name: ...}
user.save(['name', 'id']);
/ PATCH /users {name: ..., id: ...}
```

**Comment by stryju** [comment]
if that's the case, IMO it should use either array or string notation, not both - just saying :-)

**Comment by grabbou** [comment]
Yeah, maybe, then string can be left for other fancy parameters we might add at some point :D

**Comment by stryju** [comment]
well, i see nothing wrong with 

```
model.save('name');
model.save('name', 'age');
```

**Comment by grabbou** [comment]
Yeah, that's true. Let's get back to this discussion once V2 is on its way.
